main.c

```diff
@@ -686,13 +686,30 @@ int main(void)
 	gpio_set_dir(PICO_DEFAULT_LED_PIN, GPIO_OUT);
 	gpio_put(PICO_DEFAULT_LED_PIN, 1);
 
-	gpio_init(BOOTLOADER_ENTRY_PIN);
-	gpio_pull_up(BOOTLOADER_ENTRY_PIN);
-	gpio_set_dir(BOOTLOADER_ENTRY_PIN, 0);
+	struct image_header *hdr = (struct image_header *)(XIP_BASE + IMAGE_HEADER_OFFSET);
+
+	gpio_set_function(UART_TX_PIN, GPIO_FUNC_UART);
+	gpio_set_function(UART_RX_PIN, GPIO_FUNC_UART);
+	uart_init(uart0, UART_BAUD);
+	uart_set_hw_flow(uart0, false, false);
 
 	sleep_ms(10);
+	
+    while (uart_is_readable(uart0)){uart_getc(uart0);}; //clear Buffer
+    
+	//Wait loop
+    bool no_rec = true;
+
+    uart_puts(uart0, "Boot\r\n");  //Header
+
+	for(int i = 0; i < 10000; i++){
+       busy_wait_us(100);
+       if (uart_is_readable(uart0)){     
+           no_rec = false;
```

Comment:
This is not used anywhere afterwards, what is the purpose of it ?

Reply:
reflected the changes in the latest commit. it is used